### PR TITLE
Fixing multiple small typos

### PR DESCRIPTION
**Pull request by fbidu**
Hi! This is my first contribution here :D

I was using [Apiary](apiary.io) when I found a small typo in the "unrecognized block" warning message. I tracked down the typo to the parser and fixed it. Afterwards, I ran [scspell](https://github.com/myint/scspell), a lightweight spell checking tool and fixed some other typos.

I did not proofread the comments and strings where no typo was found and I tried my best not to fall in small differences between US and UK English.

### CODE REVIEW

File: src/Signature.cc [hunk 1]
```diff
@@ -42,7 +42,7 @@ SectionType snowcrash::SectionKeywordSignature(const mdp::MarkdownNodeIterator&
     TYPECHECK(Relation)
 
     /*
-     *  NOTE: Order is important. Resource MUST preceed the Action.
+     *  NOTE: Order is important. Resource MUST preced the Action.
```
Comment (kylef) [inline src/Signature.cc]: This one was correct before. [preceed](http://definefor.me/preceed) is a valid word.

File: test/test-RelationParser.cc [hunk 2]
```diff
@@ -76,7 +76,7 @@ TEST_CASE("Relation identifier starting with non lower alphabet", "[relation]")
 
 TEST_CASE("Relation identifier containing capital letters", "[relation]")
 {
-    mdp::ByteBuffer source = "+ Relation: deLete";
```
Comment (kylef) [inline test/test-RelationParser.cc]: This shouldn't be changed, this test is explicitly checking relation identifiers with uppercase characters:

```
TEST_CASE("Relation identifier containing capital letters", "[relation]")
```

File: test/test-ResourceParser.cc [hunk 3]
```diff
@@ -609,7 +609,7 @@ TEST_CASE("Parse named resource with lazy referencing with both response and req
     REQUIRE(resource.actions[0].examples[0].responses[0].reference.meta.state == Reference::StateResolved);
 }
 
-TEST_CASE("Expect to have a warning when 100 responce's reference has a body", "[resource][model]")
+TEST_CASE("Expect to have a warning when 100 response's reference has a body", "[resource][model]")
```
Comment (kylef) [inline test/test-ResourceParser.cc]: This should be `responses` not `response's`. There is no possession here.

**Comment by kylef** [comment]
Hi @fbidu, thanks for the first time contribution. I've taken a look and have some minor comments, once resolved this should be good to go.

**Comment by fbidu** [comment]
Hello @kylef, thanks for the review! I'll provide the fixes

**Comment by pksunkara** [comment]
@fbidu I am really sorry, but can you rebase your branch on master to resolve the conflicts? We did a small formatting PR merge just recently. Should've merged this first.

**Comment by fbidu** [comment]
@pksunkara yes, sure

**Comment by fbidu** [comment]
@pksunkara there you go. I solved the conflicts while honoring the new formatting

File: test/performance/perf-snowcrash.cc [hunk 4]
```diff
@@ -158,7 +158,7 @@ int main(int argc, const char* argv[])
 
     std::cout << "parsing '" << inputFileName << "' " << TestRunCount << "-times (" << result << "):\n";
     std::cout << "total: " << total << "s mean: " << mean << " +/- " << stddev << "s\n";
-
-    // FIXME: Intstrumetns helper
+
```
Comment (pksunkara) [inline test/performance/perf-snowcrash.cc]: Please delete the extra whitespace.

File: test/test-HeadersParser.cc [hunk 5]
```diff
@@ -297,8 +297,7 @@ TEST_CASE("Missing or wrong placed colon in header definition", "[headers][issue
     }
 }
 
-TEST_CASE("Allow parse nonvalid headers, provide apropriate warning", "[headers][issue][158]")
-{
+TEST_CASE("Allow parse nonvalid headers, provide appropriate warning", "[headers][issue][158]") {
```
Comment (pksunkara) [inline test/test-HeadersParser.cc]: The open bracket should be on a new line.

File: test/test-StringUtility.cc [hunk 6]
```diff
@@ -34,14 +34,14 @@ TEST_CASE("Templates for compare equality", "[utility]")
         REQUIRE_FALSE(IsIEqual()('a', 'b'));
     }
 
-    SECTION("Both version should not throw while testing diferent types")
+    SECTION("Both version should not throw while testing different types")
     {
-        REQUIRE_NOTHROW(IsEqual()('a', 1));  // allow compare diferent types - not throw
-        REQUIRE_NOTHROW(IsIEqual()('a', 1)); // allow compare diferent types - not throw
+        REQUIRE_NOTHROW(IsEqual()('a',1)); // allow compare different types - not throw
+        REQUIRE_NOTHROW(IsIEqual()('a',1)); // allow compare different types - not throw
```
Comment (pksunkara) [inline test/test-StringUtility.cc]: Spaces after commas, and align the comments please. You can see the diff above.

**Comment by fbidu** [comment]
@pksunkara oh! Thanks for noticing. Fixed them now

File: test/performance/perf-snowcrash.cc [hunk 7]
```diff
@@ -158,7 +158,6 @@ int main(int argc, const char* argv[])
 
     std::cout << "parsing '" << inputFileName << "' " << TestRunCount << "-times (" << result << "):\n";
     std::cout << "total: " << total << "s mean: " << mean << " +/- " << stddev << "s\n";
-
```
Comment (pksunkara) [inline test/performance/perf-snowcrash.cc]: This empty line was unnecessarily deleted.

File: src/snowcrash.cc [hunk 8]
```diff
@@ -79,8 +79,9 @@ int snowcrash::parse(
         std::stringstream ss;
         ss << "parser exception: '" << e.what() << "'";
         out.report.error = Error(ss.str(), ApplicationError);
-    } catch (...) {
-        out.report.error = Error("parser exception has occured", ApplicationError);
+    }
+    catch (...) {
```
Comment (pksunkara) [inline src/snowcrash.cc]: You missed this. It wasn't there before.

**Comment by fbidu** [comment]
@pksunkara done!

**Pull request merged**